esphome/components/scd30/sensor.py

```diff
@@ -47,7 +42,7 @@ def remove_altitude_suffix(value):
             ),
             cv.Optional(CONF_AUTOMATIC_SELF_CALIBRATION, default=True): cv.boolean,
             cv.Optional(CONF_ALTITUDE_COMPENSATION): cv.All(
-                remove_altitude_suffix,
+                cv.float_with_unit("altitude", "(m|m a.s.l.|m MAMSL|m MASL)"),
```

Comment:
While the regex allowed the `m`, I don't think it should be there - `meters Meters Above Sea Level` doesnt make sense
```suggestion
                cv.float_with_unit("altitude", "(m|m a.s.l.|MAMSL|MASL)"),
```

Reply:
good point!